rust-mode.el

```diff
@@ -1211,34 +1234,37 @@ whichever comes first."
                            'syntax-table (string-to-syntax "|"))
         (goto-char (match-end 0))))))
 
+
+(defvar-local rust-syntax-propertize-count 0)
 (defun rust-syntax-propertize (start end)
   "A `syntax-propertize-function' to apply properties from START to END."
-  (goto-char start)
-  (let ((str-start (rust-in-str-or-cmnt)))
-    (when str-start
-      (rust--syntax-propertize-raw-string str-start end)))
-  (funcall
-   (syntax-propertize-rules
-    ;; Character literals.
-    (rust--char-literal-rx (1 "\"") (2 "\""))
-    ;; Raw strings.
-    ("\\(r\\)#*\""
-     (0 (ignore
-         (goto-char (match-end 0))
-         (unless (save-excursion (nth 8 (syntax-ppss (match-beginning 0))))
-           (put-text-property (match-beginning 1) (match-end 1)
-                              'syntax-table (string-to-syntax "|"))
-           (rust--syntax-propertize-raw-string (match-beginning 0) end)))))
-    ("[<>]"
-     (0 (ignore
-         (when (save-match-data
-                 (save-excursion
-                   (goto-char (match-beginning 0))
-                   (rust-ordinary-lt-gt-p)))
-           (put-text-property (match-beginning 0) (match-end 0)
-                              'syntax-table (string-to-syntax "."))
-           (goto-char (match-end 0)))))))
-   (point) end))
+  (let ((rust-macro-scopes (rust-macro-scope start end)))
+    (goto-char start)
+    (let ((str-start (rust-in-str-or-cmnt)))
+      (when str-start
+        (rust--syntax-propertize-raw-string str-start end)))
+    (funcall
+     (syntax-propertize-rules
+      ;; Character literals.
+      (rust--char-literal-rx (1 "\"") (2 "\""))
+      ;; Raw strings.
+      ("\\(r\\)#*\""
+       (0 (ignore
+           (goto-char (match-end 0))
+           (unless (save-excursion (nth 8 (syntax-ppss (match-beginning 0))))
+             (put-text-property (match-beginning 1) (match-end 1)
+                                'syntax-table (string-to-syntax "|"))
+             (rust--syntax-propertize-raw-string (match-beginning 0) end)))))
+      ("[<>]"
+       (0 (ignore
+           (when (save-match-data
+                   (save-excursion
+                     (goto-char (match-beginning 0))
+                     (rust-ordinary-lt-gt-p)))
```

Comment:
start and end should be passed to rust-ordinary-lt-gt-p so that it passes them to rust-in-macro.

Reply:
It doesn't make any difference because `rust-macro-scopes` is `let`-bound. So `rust-in-macro` isn't using `start` and `end`.

Reply:
I think I saw lots of lines like "rust-syntax-propertize 300000 305000" followed by "rust-macro-scope nil nil" when I was debugging the problem above where rust-in-macro was calling rust-macro-scope repeatedly, so that's why I mentioned this, but I think yes, fixing the other issue would make this unnecessary.

(I was going through a test run that involved editing a function near the end of big.rs, and the editing was very slow until I made these couple of changes.)

Reply:
I have tested this and cannot reproduce, except in one circumstance. If I load the current master of rust-mode.el and then eval `in-macro` and `scopes`, so that the `let' binding in `rust-syntax-propertize' is not happening. Easy enough to get into this state actually, and I just spent 20 minutes debugging before I realised.

Reply:
Oh sorry about the confusion. I only saw those repetitions *before* the other fix, with the other fix I don't see them. So this was resolved properly in the other fix and you were right, this suggestion is unnecessary.